### DISCUSSION TITLE
Creating a plugin that allows me to select an arbitrary data attribute

### DISCUSSION THREAD

**Discussion by cysabi** [tailwindlabs/tailwindcss#9139] [Help]
Hello, I'd like to create some sort of plugin that allows me to specify any arbitrary data attribute and value such like this:

`&[data-state="active"]` = `<div class="data-[state=active]:bg-slate-800">`
`&[data-highlighted]` = `<div class="data-[highlighted]">`

I know that you can do `[&[data-state=active]]` but I just would like to add my own personal syntactic sugar with the plugins feature.

Is this possible?

**Comment by onmax** [comment]
So, what you want is to use one HTML element to create a CSS rule that affect another element, right?

```html

<!-- This is just to select another element. No custom CSS for this element at all -->
<div class="[data-state=\"active\"]:bg-slate-800"></div>

<!-- Elsewhere, in the dom... this will have bg-slate-800 -->
<div data-state="active"></div> 

```

**Reply by cysabi** [reply]
Not quite, I just want to create a tailwind plugin that supports arbitrary selectors

**Comment by wongjn** [comment]
I don't think there's anything like that in Tailwind. As you have already mentioned, you might as well use arbitrary variants, since it really isn't *that* different from your desired syntax conceptually.
